### DISCUSSION TITLE
For Homebrew formulas, use of fallback dependencies is disabled?

### DISCUSSION THREAD

**Discussion by andy5995** [mesonbuild/meson#9886] [General]
I was gonna update the [rmw](https://github.com/theimpossibleastronaut/rmw) homebrew  formula. After I changed the existing .rb file, I ran 

`brew install --build-from-source Formula/rmw.rb`

`brew` did its thing, and at the tail end of the output:

```
Run-time dependency canfigger found: NO (tried pkgconfig and cmake)
Not looking for a fallback subproject for the dependency canfigger because:
Use of fallback dependencies is disabled.

../meson.build:65:0: ERROR: Dependency 'canfigger' is required but not found.

A full log can be found at /tmp/rmw-20220126-46172-1m8dyep/rmw-0.8.1/build/meson-logs/meson-log.txt

Do not report this issue to Homebrew/brew or Homebrew/core!
```

I've included the source for canfigger in the rmw archive (I used 'dist --include-subprojects'). I've tested it on other systems, there's nothing wrong with how meson (or I) created the archive.

It appears that the folks at brew just don't allow the use of the fallback: keyword for some reason.

As for starting this discussion, I'm not reporting any issue with meson, but just wondering... wtf? Have any other maintainers experienced this when adding or updating their project in the [Homebrew](https://brew.sh/) repo?

**Comment by eli-schwartz** [comment]
See https://github.com/Homebrew/brew/blob/51da118a90f21c2a373307db891c113478bc249b/Library/Homebrew/formula.rb#L1561-L1565
```rb
  # Standard parameters for meson builds.
  sig { returns(T::Array[String]) }
  def std_meson_args
    ["--prefix=#{prefix}", "--libdir=#{lib}", "--buildtype=release", "--wrap-mode=nofallback"]
  end
```

And in https://github.com/Homebrew/homebrew-core/blob/4ebc58b38d2960f746579e2072b05ce46bedfef4/Formula/rmw.rb#L34 this definition is reused:
```rb
  def install
    mkdir "build" do
      system "meson", *std_meson_args, ".."
      system "ninja", "-v"
      system "ninja", "install", "-v"
    end
  end
```

It is of course the homebrew maintainers' choice to disable fallback wraps. They would probably rationalize this choice by saying it is to prevent formulas like rmw from using vendored versions of canfigger, when the homebrew way would be to add a formula for canfigger and include it as a `depends_on`.

The alternative would be to override the arguments used in the rmw formula, and specify a `--wrap-mode` that isn't the standard homebrew one.

**Comment by andy5995** [comment]
They let me just add a formula for the library (even though it's not a notable GitHub repo yet), and all is well.
